python/tests/test_coaltime_distribution.py

```diff
@@ -1016,24 +1167,39 @@ class TestSpanNormalisedCoalescenceTimeTable(TestCoalescenceTimeDistribution):
     Uniform weights on nodes summed over trees, weighted by tree span
     """
 
-    def coalescence_time_distribution(self):
+    def coalescence_time_distribution(self, mask_half_of_each_tree=False):
+        """
+        Methods should give the same result if half of each tree is masked,
+        because "span weights" are normalised using the accessible (nonmissing)
+        portion of the tree sequence.
+        """
         ts = self.ts_two_trees_four_leaves()
+        if mask_half_of_each_tree:
+            mask = np.array(
+                [
+                    [t.interval.left, (t.interval.right + t.interval.left) / 2]
+                    for t in ts.trees()
+                ]
+            )
+            ts = ts.delete_intervals(mask)
         distr = ts.coalescence_time_distribution(
             span_normalise=True,
         )
         return distr
 
     def test_weights(self):
```

Comment:
pytest parametrize is better for this sort of thing:
```python
@pytest.mark.parametrize("with_missing_data", [True, False])
def test_weights(self, with_missing_data):
     ... as before
```

Reply:
Doesn't look like you made this change @nspope?

Reply:
whoops, I made the change to the identical pattern here: https://github.com/tskit-dev/tskit/blob/604680259e9996a0d08d671c968d013999bd3381/python/tests/test_coaltime_distribution.py#L1305-L1307

I'll go through and fix